include/boost/http_proto/service/zlib_service.hpp

```diff
@@ -54,60 +50,143 @@ encoding_size_hint(decoder_config cfg = {}) noexcept
         (6 * 1024);
 }
 
+/** Error codes returned from compression/decompression functions.
+
+    @note Negative values are errors, positive values are used
+    for special but normal events.
+*/
+enum class error
+{
+    ok          =  0,
+    stream_end  =  1,
+    need_dict   =  2,
+    errno_      = -1,
+    stream_err  = -2,
+    data_err    = -3,
+    mem_err     = -4,
+    buf_err     = -5,
+    version_err = -6
+};
+
+/// Flush methods.
+enum class flush
+{
+    none,
+    partial,
+    sync,
+    full,
+    finish,
+    block,
+    trees
+};
+
+/** Input and output buffers.
+
+    @note The application must update `next_in` and `avail_in` when `avail_in`
+    has dropped to zero. It must update `next_out` and `avail_out` when
+    `avail_out` has dropped to zero.
+*/
+struct params
+{
+    /// Next input byte
+    void const* next_in;
+
+    /// Number of bytes available at `next_in`
+    std::size_t avail_in;
+
+    /// Next output byte
+    void* next_out;
+
+    /// Number of bytes remaining free at `next_out`
+    std::size_t avail_out;
+};
+
+/// Abstract interface for deflator/inflator streams.
+struct stream
```

Comment:
do you need a virtual dtor?

Reply:
No, users do not control the lifetime of the stream type. They are created and destroyed within the zlib service implementation and are only referenced by their concrete types.